lighthouse-core/lib/url-shim.js

```diff
@@ -25,4 +25,26 @@
 
 // TODO: Add back node require('url').URL parsing when bug is resolved:
 // https://github.com/GoogleChrome/lighthouse/issues/1186
-module.exports = (typeof self !== 'undefined' && self.URL) || require('whatwg-url').URL;
+const URL = module.exports = (typeof self !== 'undefined' && self.URL) || require('whatwg-url').URL;
+
+/**
+ * Safely checks if the host of a URL matches a known host with a specified
+ * fallback in case of error.
+ *
+ * @param {string} url
+ * @param {string} host
+ * @param {boolean|function=} fallback
+ * @return {boolean}
+ */
+module.exports.hostMatches = function(url, host, fallback) {
+  try {
+    return new URL(url).host === host;
+  } catch (err) {
+    let result = fallback;
+    if (typeof fallback === 'function') {
```

Comment:
this boolean/function business is definitely a bit wonky, though it doesn't seem like there's a terribly elegant way of handling the three states without a conditional in each place calling this function. It does feel like this function is trying to be a bit too clever, though, when really all we want is a way to catch bad URLs coming through the js call sites function